### ISSUE TITLE
Attribute's default_value can be modified

### ISSUE THREAD

**Issue by natesalisbury** [aws/aws-sdk-ruby-record#69]
It's possible to modify an attribute's default_value and have those modifications carry over to new instances of the model. 

Example:

```ruby
class MyModel
  include Aws::Record

  map_attr :things, default_value: {}
end
```

```ruby
> MyModel.new.things[:foo] = :bar
=> :bar
> MyModel.new.things
=> {:foo=>:bar}
```

**Comment by IbottaBaier** [comment]
That is because attribute configuration is stored at the class level.

**Comment by IbottaBaier** [comment]
You aren't really using the default value as it is intended (or at least implemented); default value is what to read if the attribute is not present -- you are treating it more like Hash.of_hashes "default value" mechanism (spawning an empty hash when nothing is present).

**Comment by natesalisbury** [comment]
Can you give me an example of when and how default_value should be used? I've read the description in the code

``` ruby
# @option opts [Object] :default_value Optional attribute used to
#   define a "default value" to be used if the attribute's value on an
#   item is nil or not set at persistence time.
```

But that still sounds to me like I could use this option to specify that I want an attribute initialized to something. 

If you're interested, I've opened a pull request here that prevents the default value from getting modified: https://github.com/aws/aws-sdk-ruby-record/pull/70

**Comment by IbottaBaier** [comment]
The default value is not like MySQL default that adds the value to the record if it is not set -- this is a read-only default value. 
```
class MyRecord
  include Aws::DynamoDB::Record
  integer_attr :id, hash_key: true
  map_attr :the_best, default_value: { cheese: "is great" }
end

record = MyRecord.new(id: 1)
record.the_best #=> { cheese: "is great" }
record.save!
record = MyRecord.find(id: 1)
record.the_best #=> nil
```

**Comment by awood45** [comment]
If we have to freeze the value, it's a bug. Reviewing the PR's suitability as a bug fix.

**Comment by awood45** [comment]
Fixed by #71

**Comment by awood45** [comment]
Released this update in version `1.1.1`
